File: ElderCareAI/server/src/main/java/com/eldercare/server/entity/HealthProfile.java
package com.eldercare.server.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Data
@Entity
@Table(name = "health_profile")
public class HealthProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private String name;

    private Integer age;
    
    private Boolean shareHealth;
    private Boolean shareDiet;
    private Boolean shareContacts;

    @ElementCollection
    private List<String> diseases;

    @ElementCollection
    private List<String> allergies;

    private Long updatedAt;
}

File: ElderCareAI/app/src/main/java/com/eldercare/ai/data/entity/HealthProfile.kt
package com.eldercare.ai.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 用户健康档案实体
 * 用于健康规则匹配、子女远程设置饮食禁忌
 */
@Entity(tableName = "health_profile")
data class HealthProfile(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,

    /** 姓名 */
    val name: String = "",

    /** 年龄 */
    val age: Int = 0,

    /** 疾病/慢病：高血压、糖尿病等，逗号或列表 */
    val diseases: List<String> = emptyList(),

    /** 过敏/禁忌：不能吃花生、海鲜等 */
    val allergies: List<String> = emptyList(),

    /** 更新时间 */
    val updatedAt: Long = System.currentTimeMillis()
)
